### ISSUE TITLE
Cernan tracking deleted files

### ISSUE THREAD

**Issue by dvdklnr** [postmates/cernan#244]
Here's an lsof output excerpt:
```
COMMAND   PID USER   FD   TYPE  DEVICE   SIZE/OFF     NODE NAME
cernan     3877  3912         root   19r      REG              202,1     337348     149000 /var/log/upstart/consul-agent.log.1 (deleted)
cernan     3877  3912         root   34r      REG              202,1        398     149073 /var/log/upstart/oxingest-common.log.1 (deleted)
cernan     3877  3912         root   35r      REG              202,1 1445977984     149017 /var/log/upstart/viper.log.1 (deleted)
cernan     3877  3912         root   41r      REG              202,1        117     149198 /var/log/upstart/cernan.log.1 (deleted)```

**Comment by blt** [comment]
@dvdklnr I _believe_ this is largely resolved with #252. It's still possible for the file sink to hold references to paths that have been deleted but to a lesser extent than before. 

The glob loop keeps a BTreeMap of path -> file pointer. Right now the glob loop only _adds_ to the map, meaning if a path disappears we'll hold that file pointer forever. This is resolvable by checking the existence of the path on every glob loop reset, once every 60s. 

I'll update this issue when that work is done.

**Comment by blt** [comment]
Okey dokey, here's what we've got. As mentioned on April 13 we keep that hashmap of path -> file pointer around and occasionally add to it. That occurs here: https://github.com/postmates/cernan/blob/b2236f0dcebaff4fad18de39994da3613dfda9e0/src/source/file.rs#L247

Now, what we can do is, when we're examining this table, is remove files that have gone "too long" since their last update _and_ are no longer the descriptor we find when we pull that from the path. We're for sure going to make "too long" a configurable option. Now, this opens us up to the risk of losing a file which is very slow. Say "too long" is set at 15 minutes and we have a log file which gets a new line once every 30 minutes. Cernan, so configured, will drop the descriptor from its polling map and add it to some kind of ignore map, probably path -> file pointer. Until the file rolls over anything added to the ignored file will be lost. 

That... isn't great. I'm open to other suggestions. 

(For fast log files –– where roll-over happens -- we _do_ drop the descriptor and so, over time, the OS should clean them up. I'll do some experiments to confirm that this is working as expected.)

@dvdklnr @lynchc @tsantero @blakebarnett
